client-course-schedulizer/src/components/reuseables/AddSectionPopover/AddSectionPopover.tsx

```diff
@@ -104,53 +121,60 @@ export const AddSectionPopover = () => {
           duration: Number(data.duration),
           location: {
             building: location[0],
+            roomCapacity: Number(data.roomCapacity),
             roomNumber: location[1],
           },
-          startTime: data.startTime,
+          startTime: startTimeCase(data.startTime),
         },
       ],
       semesterLength: semesterType,
+      studentHours: Number(data.studentHours),
       term: data.term,
       year: "2021-2022",
     };
 
-    // TODO: Append section to previously existing course if a course has already been created
     const newCourse: Course = {
       facultyHours: Number(data.facultyHours),
       name: data.name,
       number: data.number,
-      prefixes: [data.prefix],
-      sections: [newSection],
+      prefixes: prefixCase(data.prefix),
+      // The newSection will be added later in insertSectionCourse()
+      sections: [],
       studentHours: Number(data.studentHours),
     };
-    // eslint-disable-next-line no-console
-    return console.log(newCourse);
+
+    // Insert the Section to the Schedule, either as a new Course or to an existing Course
+    insertSectionCourse(schedule, newSection, newCourse);
+
+    appDispatch({ payload: { schedule }, type: "setScheduleData" });
```

Comment:
Maybe await this dispatch? I know we had an issue with not awaiting on a different PR, but I'm still confused as to why

Reply:
When I tried this (like `await appDispatch({ payload: { schedule }, type: "setScheduleData" });`), I got a warning saying: `'await' has no effect on the type of this expression. ts(80007)`, so I guess this isn't necessary? Feel free to correct me if I'm misunderstanding.